cmake/configure/configure_2_trilinos.cmake

```diff
@@ -42,7 +42,7 @@ MACRO(FEATURE_TRILINOS_FIND_EXTERNAL var)
       )
 
     FOREACH(_module
-      Amesos Epetra Ifpack AztecOO Sacado Teuchos ML
+      Amesos Epetra Ifpack AztecOO Sacado Teuchos ML ROL
```

Comment:
This means we would require at least `Trilinos-12.4`.

Reply:
i guess we need a way to get optional modules. @tamiko ping.

Reply:
and have something like `DEAL_II_TRILINOS_WITH_ROL`

Reply:
@vishalkenchan i suggest move `rol` into a separate `FOREACH`, you can put it right above, something along the lines (not tested!)
```
FOREACH(_optional_module
     ROL
     )
     ITEM_MATCHES(_module_found ${_optional_module} ${Trilinos_PACKAGE_LIST})
     SET(DEAL_II_TRILINOS_WITH_${_optional_module} ${_module_found})
ENDFOREACH()
```
then you should be able to guard your adapters with `DEAL_II_TRILINOS_WITH_ROL`.